### PR TITLE
Allow test users to use chatbot

### PR DESCRIPTION
**Pull request by TamiTakamiya**
Jira Issue: <https://issues.redhat.com/browse/AAP-34721>

## Description

Allow test users to use chatbot in addition to RH employees so that we can start writing E2E/Performance test cases.

### Steps to test
1. Pull down the PR
2. Run unit tests

### Scenarios tested

Unit tests are included in the PR

## Production deployment

- [X] This code change is ready for production on its own
- [ ] This code change requires the following considerations before going to production:

### CODE REVIEW

File: ansible_ai_connect/main/permissions.py [hunk 1]
```diff
@@ -28,4 +29,4 @@ def has_permission(self, request, view):
             return permitted
 
         user = request.user
-        return user.rh_employee
+        return user.rh_employee or "test" in list(user.groups.values_list("name", flat=True))
```
Comment (jameswnl) [inline ansible_ai_connect/main/permissions.py:32]: you can also use this expression to check group membership:
`user.groups.filter(name="test").exists()` 
which was used previously in https://github.com/ansible/ansible-ai-connect-service/pull/1341/files#diff-d8b9627bf3e9cc38433bebfcb64ea46bc5dee93c7eaa4e4fc62fe6c3fa1d4341

File: ansible_ai_connect/main/views.py [hunk 2]
```diff
@@ -115,7 +115,7 @@ class ChatbotView(ProtectedTemplateView):
     template_name = "chatbot/index.html"
 
     permission_classes = [
-        IsAuthenticatedRHEmployee,
+        IsAuthenticatedRHEmployeeOrTestUser,
```
Comment (jameswnl) [inline ansible_ai_connect/main/views.py:118]: Please create this `TestUser` as a separate permission object and use bitwise `|` like https://stackoverflow.com/questions/35557156/how-to-get-or-permissions-instead-of-and-in-rest-framework

**Review by jameswnl — CHANGES_REQUESTED** [review]

File: ansible_ai_connect/main/permissions.py (hunk 1 above)
Comment (TamiTakamiya) [reply ansible_ai_connect/main/permissions.py:32]: Oh, that's neat. I'll update the code. Thanks!

File: ansible_ai_connect/main/views.py (hunk 2 above)
Comment (TamiTakamiya) [reply ansible_ai_connect/main/views.py:118]: Yes. That's cleaner. Thanks.

File: ansible_ai_connect/users/views.py [hunk 3]
```diff
@@ -88,7 +88,12 @@ def get_context_data(self, **kwargs):
         context["documentation_url"] = settings.COMMERCIAL_DOCUMENTATION_URL
 
         user = self.request.user
-        context["rh_employee"] = user.rh_employee if isinstance(user, User) else False
+        if isinstance(user, User):
+            context["rh_employee_or_test_user"] = user.rh_employee or "test" in list(
+                user.groups.values_list("name", flat=True)
```
Comment (jameswnl) [inline ansible_ai_connect/users/views.py:93]: Can you update this expression to be the same as the above one?

**Review by jameswnl — APPROVED** [review]

File: ansible_ai_connect/main/permissions.py [hunk 4]
```diff
@@ -11,21 +11,30 @@
...
+    Allow access only to test users, who are found in the "test" Django group
+    """
+
+    code = "permission_denied_user_not_test_user"
+    message = "The user is not a Test User"
+
+    def has_permission(self, request, view):
         user = request.user
-        return user.rh_employee
+        return user.is_authenticated and user.groups.filter(name="test").exists()
```
Comment (goneri) [inline ansible_ai_connect/main/permissions.py:40]: The condition is wrong, by returning `False` if the user has no `test` group, it prevents the rh_employee users from connecting.

File: ansible_ai_connect/main/views.py [hunk 5]
```diff
@@ -115,7 +115,7 @@ class ChatbotView(ProtectedTemplateView):
     template_name = "chatbot/index.html"
 
     permission_classes = [
-        IsAuthenticatedRHEmployee,
+        IsRHEmployee | IsTestUser,
```
Comment (jameswnl) [inline ansible_ai_connect/main/views.py:118]: @goneri this `|` condition

File: ansible_ai_connect/main/permissions.py (hunk 4 above)
Comment (goneri) [reply ansible_ai_connect/main/permissions.py:40]: update: I'm wrong, cf. @jameswnl comments below.

File: ansible_ai_connect/users/templates/users/home.html [hunk 6]
```diff
@@ -129,7 +129,7 @@ <h1 class="pf-c-title pf-m-lg">{{ project_name }}</h1>
                   {% if deployment_mode == 'saas' and user.is_authenticated and user.rh_user_is_org_admin %}
                   <a class="pf-l-level__item" href="/console"><span class="fas fa-solid fa-cog"></span> Admin Portal</a>
                   {% endif %}
-                  {% if chatbot_enabled and deployment_mode == 'saas' and user.is_authenticated and user.rh_employee %}
+                  {% if chatbot_enabled and deployment_mode == 'saas' and user.is_authenticated and rh_employee_or_test_user %}
```
Comment (goneri) [inline ansible_ai_connect/users/templates/users/home.html:132]: You also need to adjust https://github.com/ansible/ansible-ai-connect-service/blob/main/ansible_ai_connect/users/templates/users/trial.html#L134-L136

**Pull request merged**